lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -682,6 +682,11 @@ class TypeRefinementContextBuilder : private ASTWalker {
     if (isCurrentTRCContainedByDeploymentTarget())
       return false;
 
+    // A declaration inside of a local context always inherits the availability
+    // of the parent.
+    if (D->getDeclContext()->isLocalContext())
```

Comment:
Are there any local declarations that can have an explicit `@available` attribute of their own? If so, will we handle them correctly?

Reply:
This won't prevent explicit availability attributes from taking effect. Rather, it prevents the assumption that the signature of the declaration has an implicit availability of the deployment target.